pkg/controller/cluster/cluster.go

```diff
@@ -279,6 +287,56 @@ func (c *ClusterReconciler) server(ctx context.Context, cluster *v1alpha1.Cluste
 	return nil
 }
 
+func (c *ClusterReconciler) bindNodeProxyClusterRole(ctx context.Context, cluster *v1alpha1.Cluster) error {
```

Comment:
I dont think this should be added here, this is a part of the resources related to the shared agent, it should be added in the resources function of the shared agent implementation https://github.com/rancher/k3k/blob/main/pkg/controller/cluster/agent/shared.go#L67

Reply:
Yes, it's what I initially did but unfortunately it doesn't work, I forgot to add a comment about this. Basically the ClusterRole is a Cluster resource, and the Controller is Namespaced. That means it cannot create namespaced resource, and it fails with the following error: `cluster-scoped resource must not have a namespace-scoped owner`. :slightly_frowning_face: 

The only solution I found was to create the ClusterRole and the Binding during the Helm installation, and add/remove the subjects during the cluster creation/deletion.

Reply:
hmmm, interesting, thats fine then, should we have a check for the virtual mode vs shared mode, since we dont need to invoke the function in the agent mode